File: C++/conversiónMillasAKilometros.cpp
/*Conversión de unidades: Convierte kilómetros a millas y viceversa.*/

#include <iostream>
using namespace std;

double millas(double ingresarMillas)
{
    float valorKilometros = 1.60934;
    cout << "ingresa millas: ";
    cin >> ingresarMillas;
    return ingresarMillas * valorKilometros;
}

double kilometros(double ingresaKilometrso)
{
    float valorMillas = 0.621371;
    cout << "Imgresa kilometros: ";
    cin >> ingresaKilometrso;
    return ingresaKilometrso * valorMillas;
}

int main()
{
    int opcion;
    float Millas, Kilometros, resultado;
    /*float valorKilometros = 1.60934;*/
    /*float valorMillas = 0.621371;*/

    cout << "1. Millas a Kilometros." << endl;
    cout << "2. Kilometros a Millas." << endl;
    cout << "Elige entre 1 - 2: ";
    cin >> opcion;

    switch (opcion)
    {
    case 1:
        cout << "MILLAS A KILOMETROS." << endl;
        resultado = millas(Millas);
        cout << resultado;
        /*
        cout << "Ingresa Millas: ";
        cin >> millas;
        resultado = millas * valorKilometros;
        cout << "Distancia a kilometros: " << resultado;
        */
        break;

    case 2:
        cout << "KILOMETROS A MILLAS." << endl;
        resultado = kilometros(Kilometros);
        cout << resultado;
        /*
        cout << "Ingresa kilometros: ";
        cin >> kilometros;
        resultado = kilometros * valorMillas;
        cout << "Distancia a Millas: " << resultado;
        */
        break;

    default:
        cout << "ERROR - Debes elegir entre 1 - 2";
        break;
    }
}

File: JavaScript/conversiónMillasAKilometros.js
function millas(ingresarLasMillas) {
    /* Los prompt se deben poner fuera de las funciones */
    let valoKilometro = 1.60934
    return ingresarLasMillas * valoKilometro
}

function Kilometros(ingresarLosKilometros) {
    let valoMillas = 0.621371
    
    return ingresarLosKilometros * valoMillas
}

let opcion = parseInt(prompt("1. Millas a Kilometros\n2. Kilometros a Millas\nElige entre 1 - 2: "))
let millasIngresadas, kilometrosIngresadas
/*let valoKilometro = 1.60934
let valorMilla = 0.621371
*/
let conversion = 0


switch (opcion) {
    case 1:
        document.write("ELEGISTE MILLAS A KILOMETROS <br>")
        let millasIngresadas = parseFloat(prompt("Ingresa tu recorrido en Millas: "))
        conversion = millas(millasIngresadas)
        document.write (`La conversión a kilometros es: ${conversion}`)

        /*
        let millas = parseFloat(prompt("Ingresa tu recorrido en Millas: "))
        conversion = millas * valoKilometro
        document.write(`Millas ${millas} conversión a Kilometros es ${conversion} km`)
        */

        break;
    case 2:
        document.write("ELEGISTE KILOMETROS A MILLAS <br>")
        let kilometrosIngresadas = parseFloat(prompt("Ingresa tu recorrido en Kilometros: "))
        conversion = Kilometros(kilometrosIngresadas)
        document.write(`La conversión a Millas es: ${conversion}`)

        /*
        let Kilometros = parseFloat(prompt("Ingresa tu recorrido en Millas: "))
        conversion = Kilometros * valorMilla
        document.write(`Kilometros ${Kilometros} conversión a Millas es ${conversion} millas`)
        */

        break;

    default:
        document.write("ERROR - Debes ingresa")
        break;
}